Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -23,17 +26,48 @@
 #include "MantidKernel/ListValidator.h"
 #include "MantidKernel/Material.h"
 #include "MantidKernel/MersenneTwister.h"
+#include "MantidKernel/PhysicalConstants.h"
+#include "MantidKernel/VectorHelper.h"
 
 using namespace Mantid::API;
 using namespace Mantid::Kernel;
 using Mantid::DataObjects::Workspace2D;
+namespace PhysicalConstants = Mantid::PhysicalConstants;
 
 namespace {
 constexpr int DEFAULT_NPATHS = 1000;
 constexpr int DEFAULT_SEED = 123456789;
 constexpr int DEFAULT_NSCATTERINGS = 2;
 constexpr int DEFAULT_LATITUDINAL_DETS = 5;
 constexpr int DEFAULT_LONGITUDINAL_DETS = 10;
+
+/// Energy (meV) to wavevector (angstroms-1)
+inline double toWaveVector(double energy) { return sqrt(energy / PhysicalConstants::E_mev_toNeutronWavenumberSq); }
+
+/// wavevector (angstroms-1) to Energy (meV)
+inline double fromWaveVector(double wavevector) {
+  return PhysicalConstants::E_mev_toNeutronWavenumberSq * wavevector * wavevector;
+}
+
+struct EFixedProvider {
+  explicit EFixedProvider(const ExperimentInfo &expt) : m_expt(expt), m_emode(expt.getEMode()), m_value(0.0) {
+    if (m_emode == DeltaEMode::Direct) {
+      m_value = m_expt.getEFixed();
+    }
+  }
+  inline DeltaEMode::Type emode() const { return m_emode; }
+  inline double value(const Mantid::detid_t detID) const {
+    if (m_emode != DeltaEMode::Indirect)
+      return m_value;
+    else
+      return m_expt.getEFixed(detID);
+  }
+
+private:
+  const ExperimentInfo &m_expt;
+  const DeltaEMode::Type m_emode;
+  double m_value;
```

Comment:
what is `m_value` here? looks like something to do with the energy mode?

Reply:
This struct\class is also copied from MonteCarloAbsorption.cpp. `m_value` is the EFixed value on a direct instrument. So for an elastic instrument or an indirect (where it seems the EFixed value can potentially vary by detector) it's not used. Naming could be improved by calling it m_EFixed or similar. I'll update it. In an ideal world perhaps this bit of code could also be shared between the two algorithms but since it's only ~10 lines it's perhaps not worth having a separate .h\.cpp file

Reply:
Thanks, `m_EFixed` sounds good to me - agreed that ~10 lines is not worth a separate file!

Reply:
For Indirect EFixed only depends on the analyser & reflection. It does not vary by detector.